pyro/contrib/timeseries/gp.py

```diff
@@ -86,11 +92,11 @@ def _forecast(self, dts, filtering_state, include_observation_noise=True):
         """
         assert dts.dim() == 1
         dts = dts.unsqueeze(-1).unsqueeze(-1).unsqueeze(-1)
-        trans_mat = self.kernel.transition_matrix(dts)
-        predicted_mean = torch.matmul(filtering_state.loc.unsqueeze(-2), trans_mat).squeeze(-2)[..., 0]
-        predicted_function_covar = torch.matmul(trans_mat.transpose(-1, -2),
+        trans_matrix, process_covar = self.kernel.transition_matrix_and_covariance(dt=dts)
+        predicted_mean = torch.matmul(filtering_state.loc.unsqueeze(-2), trans_matrix).squeeze(-2)[..., 0]
+        predicted_function_covar = torch.matmul(trans_matrix.transpose(-1, -2),
                                                 torch.matmul(filtering_state.covariance_matrix,
-                                                trans_mat))[..., 0, 0]
+                                                trans_matrix))[..., 0, 0] + process_covar[..., 0, 0]
```

Comment:
add dropped term

Reply:
1. Is this covered by a test?

2. It seems wasteful to compute matmuls and then index. Is it worth optimizing, say as
```py
torch.einsum("...a,...b,...ab->...",
             trans_matrix[..., 0], trans_matrix[..., 0],
             filtering_state.covariance_matrix)
```

Reply:
1. this is partially tested by the "assert monotonic increase of predictive noise" test

Reply:
ok let me try the einsum